qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -174,22 +181,24 @@ def _eval_aux_operators(
                 else:
                     value = StateFn(operator, is_measurement=True).eval(wavefn)
                 value = value.real if abs(value.real) > threshold else 0.0
-            values.append((value, 0))
-        return np.array(values, dtype=object)
+            values[key] = (value, 0)
+        return values
 
     def compute_eigenvalues(
-        self, operator: OperatorBase, aux_operators: Optional[List[Optional[OperatorBase]]] = None
+        self, operator: OperatorBase, aux_operators: Optional[ListOrDict[OperatorBase]] = None
     ) -> EigensolverResult:
         super().compute_eigenvalues(operator, aux_operators)
 
         if operator is None:
             raise AlgorithmError("Operator was never provided")
 
         self._check_set_k(operator)
-        if aux_operators:
+        if isinstance(aux_operators, list) and len(aux_operators) > 0:
             zero_op = I.tensorpower(operator.num_qubits) * 0.0
             # For some reason Chemistry passes aux_ops with 0 qubits and paulis sometimes.
             aux_operators = [zero_op if op == 0 else op for op in aux_operators]
+        elif isinstance(aux_operators, dict) and len(aux_operators) > 0:
+            aux_operators = {key: op for key, op in aux_operators.items() if not op}
```

Comment:
Why do check `if not op` at the end? Shouldn't this be `if op`? Or am I misunderstanding something here?

Reply:
Well spotted, the `not` should not be there